### PR TITLE
doc: add Sphinx site and publish it through GitHub Pages

### PR DESCRIPTION
**Pull request by umarcor**
As proposed in https://github.com/trabucayre/openFPGALoader/pull/102#issuecomment-894667089, this PR converts documentation sources from markdown to restructuredtext, and the configuration files for an Sphinx site are added. A job in the CI workflow builds the site and pushes it to branch `gh-pages`, so that it is served at trabucayre.github.io/openFPGALoader. That works for forks as well. See, for instance, [umarcor.github.io/openFPGALoader](https://umarcor.github.io/openFPGALoader).

The main reason for using Sphinx, apart from interpshinx, is being able to enhance the documentation with timing diagrams through wavedrom. See https://wishbone-interconnect.readthedocs.io/en/latest/03_classic.html#reset-operation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,42 +9,40 @@
 </p>
 
 <p align="center">
-  <strong><a href="doc/first-steps.md">First steps</a> • <a href="INSTALL.md">Install</a> • <a href="doc/troubleshooting.md">Troubleshooting</a></strong> • <a href="doc/advanced-usage.md">Advanced usage</a>
+  <strong><a href="https://trabucayre.github.io/openFPGALoader/guide/first-steps.html">First steps</a> • <a href="https://trabucayre.github.io/openFPGALoader/guide/install.html">Install</a> • <a href="https://trabucayre.github.io/openFPGALoader/guide/troubleshooting.html">Troubleshooting</a></strong> • <a href="https://trabucayre.github.io/openFPGALoader/guide/advanced.html">Advanced usage</a>
 </p>
 
-Universal utility for programming FPGAs. Compatible with many boards, cables and FPGA from major manufacturers (Xilinx, Altera/Intel, Lattice, Gowin, Efinix, Anlogic). openFPGALoader works on [Linux](INSTALL.md#linux), [Windows](INSTALL.md#windows) and [macOS](INSTALL.md#macos).
+Universal utility for programming FPGAs. Compatible with many boards, cables and FPGA from major manufacturers (Xilinx, Altera/Intel, Lattice, Gowin, Efinix, Anlogic). openFPGALoader works on Linux, Windows and macOS.
```
Comment (trabucayre) [inline README.md:15]: Maybe not really important, but why not keeping links?

**Comment by trabucayre** [comment]
Awesome!
LGTM!

**Comment by trabucayre** [comment]
Applied
Thanks!

File: README.md (hunk 1 above)
Comment (umarcor) [reply README.md:15]: I found it redundant given there is a visible sidebar in [trabucayre.github.io/openFPGALoader/guide/install](https://trabucayre.github.io/openFPGALoader/guide/install):

image

Nevertheless, I can bring them back if you want.

**Comment by umarcor** [comment]
@trabucayre note that you can edit the fields in the sidebar of the root of the repo, in order to show the URL there:

image

See, for instance, https://github.com/edaa-org/pyEDAA.ProjectModel.

**Comment by trabucayre** [comment]
Sidebard updated ! Thanks
For links in README: it's true for github.io it's redundant, it's for the github repo's main-page where this info may be more interesting. But since there is link to the install instructions page and github.io url so it's maybe no more required.
Thanks again for your work!

**Pull request merged**